apis/kueue/webhooks/clusterqueue_webhook.go

```diff
@@ -156,6 +156,7 @@ func validateFlavorQuotas(flavorQuotas kueue.FlavorQuotas, coveredResources []co
 	allErrs := validateNameReference(string(flavorQuotas.Name), path.Child("name"))
 	if len(flavorQuotas.Resources) != len(coveredResources) {
 		allErrs = append(allErrs, field.Invalid(path.Child("resources"), field.OmitValueType{}, "must have the same number of resources as the coveredResources"))
+		return allErrs
 	}
 
 	for i, rq := range flavorQuotas.Resources {
```

Comment:
```suggestion
	}

	coveredResNum := len(coveredResources)
	for i, rq := range flavorQuotas.Resources {
		if coveredResNum < i+1 {
			break
		}
```

Validating matched resources would be helpful.

Reply:
Done.